src/Features/LanguageServer/Protocol/Handler/CodeLens/CodeLensHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeLens;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(CodeLensHandler)), Shared]
+[Method(LSP.Methods.TextDocumentCodeLensName)]
+internal class CodeLensHandler : ILspServiceDocumentRequestHandler<LSP.CodeLensParams, LSP.CodeLens[]?>
```

Comment:
```suggestion
internal sealed class CodeLensHandler : ILspServiceDocumentRequestHandler<LSP.CodeLensParams, LSP.CodeLens[]?>
```

Reply:
nit: consider calling this LspCodeLensHandler to make it more obvious this isn't hte normal code lens code.  But i also don't care much, and if this is the naming pattern here, feel free to keep.

Reply:
None of the other handlers are named LSP - I think its reasonable to let that be taken care of by the namespace + folder structure its in